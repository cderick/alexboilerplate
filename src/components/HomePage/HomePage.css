.mission {
    color: rgb(118, 115, 118);
    font-size: 29px;
    font-family: Raleway-Light;
    text-align: center;
    letter-spacing: 0.47px;
    line-height: 50px;
    position: relative;
}

.missionBack {
    background: rgb(245, 245, 245);
}

.tonnesText {
    color: rgb(118, 115, 118);
    font-size: 24px;
    font-family: Raleway-Light;
    text-align: center;
    letter-spacing: 0.34px;
    line-height: 33px;
}