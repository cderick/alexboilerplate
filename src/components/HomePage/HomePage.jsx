import React from 'react'
import style from './HomePage.scss'
import homePageSection from './data'
import ContactUsSection from './ContactUsSection/ContactUsSection'
import HomeMissionSection from './HomeMissionSection/HomeMissionSection'
import HomeBenefitSection from './HomeBenefitSection/HomeBenefitSection'


class HomePage extends React.Component {
	render() {
		const { heroBanner, howWeDoSection, listingSection, sectorSection, benefitSection, contactUs } = homePageSection
		if(!homePageSection) return null
		return (
			<section>
				<div className={`container-fluid ${style.missionBack} pl-0 pr-0 pb-5`}>
					<div className="container p-0">
						<HomeMissionSection heroBanner={heroBanner} howWeDoSection={howWeDoSection} />
						<HomeBenefitSection listingSection={listingSection} sectorSection={sectorSection} benefitSection={benefitSection} />
					</div>
				</div>
				<ContactUsSection contactUs={contactUs} />
			</section>

		)
	}
}

export default HomePage
