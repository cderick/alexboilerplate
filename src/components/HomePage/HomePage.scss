.missionBack {
    background: rgb(245, 245, 245);
}

.titleBackground {
    border-radius: 0 67.5px 67.5px 0;
    background: rgb(81, 123, 136)
}

.backgroundRight {
    border-radius: 67.5px 0 0 67.5px;
    background: rgb(81, 123, 136)
}

.strongText {
    color: rgb(81, 123, 136);
}

.boxTitleColor {
    color: rgb(99, 42, 68);
    min-height: 95px;
}

.wrapBox {
    min-height: 95px;
}

.butts {
    background: rgb(105, 175, 227);
    padding: 10px 35px;
    border-radius: 25px;
    cursor: pointer;
    width: 200px;
    color: white;
    text-align: center;
}

.myUl {
    li {
        color: #69afe3
    }
}

.roundBlue {
    background-repeat: no-repeat;
    background-size: 100%;
    background-position: center center;
    position: relative;
    min-height: 138px;
    p {
        text-align: center;
        position: absolute;
        margin-left: auto;
        margin-right: auto;
        left: 0;
        right: 0;
        top: 41%;
        color: white;
    }
    &.construction {
        background-image: url('../../static/images/Construction.svg');
    }
    &.hotels {
        background-image: url('../../static/images/Hotels.svg');
    }
    &.offices {
        background-image: url('../../static/images/Offices.svg');
    }
    &.medical {
        background-image: url('../../static/images/Medical.svg');
    }
    &.restaurants {
        background-image: url('../../static/images/Restaurants.svg');
    }
    &.retail {
        background-image: url('../../static/images/Retail.svg');
    }
}