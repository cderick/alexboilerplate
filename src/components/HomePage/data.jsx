const homePageSection = {
    heroBanner: {
        title: 'Our mission at Globechain',
        subtitle: 'Is to divert 100 million tonnes of waste from landfill by 2025 and contribute to a better world.',
        description: 'So far we have diverted 6.1 million tonnes and saved charities over £2 million while helping numerous communities across the world.',
        logoImage: `logowhite.png`
    },
    howWeDoSection: [
        {
            title: 'How do we do this?',
            titleType: 'button',
            description: 'Our marketplace allows anyone to list items that they no longer need to be reused by someone else. Our members get to request and collect the items for FREE!'
        },
        {
            class: 'float-right',
            title: 'Who is our marketplace for?',
            titleType: 'button',
            description: 'Charities, corporates, local communities, small businesses, shops, construction sites, hospitals and even friends and family.',
            shout: 'Everyone!'
        },
        {
            title: 'What does it cost?',
            titleType: 'button',
            description: 'It is free for anyone to register and request items. We charge to list items and we have the following options listed below.'
        }
    ],
    listingSection: [
        {
            title: 'Pay as you go',
            subtitle: 'For one off and ad hoc listings',
            description: '£10 per listing of up to 100 of the same item (eg. £10 to list 100 chairs)',
            blueButton: 'Sign up'

        },
        {
            title: 'Annual membership',
            subtitle: 'For multiple and unlimited listings',
            description: 'Bespoke quote based on requirements',
            blueButton: `Let's talk`

        },
        {
            title: 'Internal Reuse',
            subtitle: 'A global solution for companies to share internally',
            description: 'Bespoke quote based on requirements',
            blueButton: `Let's talk`

        }
    ],
    sectorSection: {
        title: 'Sectors',
        subtitle: 'We work with anyone from small business, large corporates to global enterprises',
        circles: [{
            type: 'Construction',
            stylename: 'construction'
        },
        {
            type: 'Hotels',
            stylename: 'hotels'
        },
        {
            type: 'Offices',
            stylename: 'offices'
        },
        {
            type: 'Medical',
            stylename: 'medical'
        },
        {
            type: 'Restaurants',
            stylename: 'restaurants'
        },
        {
            type: 'Retail',
            stylename: 'retail'
        }
    ],
    },
    benefitSection: {
        title: 'Benefits',
        list: [
            'Save money by minimizing waste disposal costs',
            'Make a social impact',
            'Optimise stock usage and inventory',
            'Downloadable data reports',
            'ESG data (Economic, Social, Governance data)',
            'Dedicated support',
            'Multiple accounts under one company'
        ]
    },
    contactUs: {
        title: 'Contact Us',
        subTitle: 'Have a question or need some help?',
        description: 'Please fill in the form below and we will be in touch!',
        forms: [
            {
                type: 'text',
                value: 'contactName',
                placeHolder: 'Name',
                required: true,
            },
            {
                type: 'text',
                value: 'contactCompany',
                placeHolder: 'Company',
            },
            {
                type: 'email',
                value: 'contactEmail',
                placeHolder: 'Email Address',
                required: true,
            },
            {
                type: 'text',
                value: 'contactPhoneNumber',
                placeHolder: 'Telephone Number',
            },
            {
                type: 'select',
                value: 'contactEnquiry',
                placeHolder: 'Enquiry',
                options: [
                    'Enquiry',
                    'Account queries',
                    'Careers',
                    'Complaints / reporting members',
                    'Feedback',
                    'I am interested in being an Ambassador',
                    'Partnership opportunities',
                    'Press / media enquiries',
                    'Something exciting',
                ],
            },
            {
                type: 'textarea',
                value: 'contactMessage',
                placeHolder: 'Type your message here',
            },
            {
                type: 'button',
                value: 'contactSendbutton',
                placeHolder: 'Send',
            },
        ],
    }
}

export default homePageSection