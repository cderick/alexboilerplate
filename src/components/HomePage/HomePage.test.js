import React from 'react'
import renderer from 'react-test-renderer'
import HomePage from './HomePage'
import { MockedProvider } from '@apollo/react-testing'


describe('Test on <HomePage /> container', () => {
  test('should render with no props', () => {
    const componentNoprops = renderer.create(
      <MockedProvider mocks={[]}>
        <HomePage />
      </MockedProvider>,
    )
    const treeNoprops = componentNoprops.toJSON()
    expect(treeNoprops).toMatchSnapshot()
  })
})

