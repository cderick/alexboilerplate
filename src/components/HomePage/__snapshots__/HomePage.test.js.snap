// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <HomePage /> container should render with no props 1`] = `
<section>
  <p>
    Hey
  </p>
</section>
`;
